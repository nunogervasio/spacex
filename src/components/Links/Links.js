import React from "react";
import styles from "./Links.module.css";
import PropTypes from "prop-types";

const Links = ({ youtube_url, youtubeImg, wiki_url, wikiImg }) => {
  return (
    <div className={styles.links}>
      <div>
        <a href={youtube_url} target="_blank" rel="noopener noreferrer">
          <img
            src={youtubeImg}
            alt="youtube icon"
            className={styles.float_left}
          />
          <span className={styles.youtube}>YouTube Video</span>
        </a>
      </div>
      <div>
        <a href={wiki_url} target="_blank" rel="noopener noreferrer">
          <img src={wikiImg} alt="wiki icon" className={styles.float_left} />
          <span className={styles.wiki}>Wikipedia Page</span>
        </a>
      </div>
    </div>
  );
};

export default Links;

Links.propTypes = {
  youtube_url: PropTypes.string.isRequired,
  youtubeImg: PropTypes.string.isRequired,
  wiki_url: PropTypes.string.isRequired,
  wikiImg: PropTypes.string.isRequired
};
