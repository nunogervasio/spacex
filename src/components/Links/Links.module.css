.links {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  margin-bottom: 1em;
  padding: 1em;
}
.links img {
  max-height: 20px;
  max-width: 20px;
}
/****** floated columns */
.youtube,
.wiki {
  float: right;
  margin-left: 5px;
}

.float_left {
  width: 25%;
  float: left;
}
