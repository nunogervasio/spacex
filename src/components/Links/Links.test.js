import React from "react";
import { shallow } from "enzyme";
import Links from "./Links";

describe("Links", () => {
  const props = {
    youtube_url: "",
    youtubeImg: "",
    wiki_url: "",
    wikiImg: ""
  };
  const links = shallow(<Links {...props} />);

  it("renders properly", () => {
    expect(links).toMatchSnapshot();
  });

  it("anchor tag links to url", () => {
    //   TODO
  });
});
