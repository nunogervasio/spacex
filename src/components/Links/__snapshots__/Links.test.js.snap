// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Links renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Links
    wikiImg=""
    wiki_url=""
    youtubeImg=""
    youtube_url=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <a
            href=""
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="youtube icon"
              className="float_left"
              src=""
            />
            <span
              className="youtube"
            >
              YouTube Video
            </span>
          </a>
        </div>,
        <div>
          <a
            href=""
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="wiki icon"
              className="float_left"
              src=""
            />
            <span
              className="wiki"
            >
              Wikipedia Page
            </span>
          </a>
        </div>,
      ],
      "className": "links",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href=""
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="youtube icon"
              className="float_left"
              src=""
            />
            <span
              className="youtube"
            >
              YouTube Video
            </span>
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="youtube icon"
                className="float_left"
                src=""
              />,
              <span
                className="youtube"
              >
                YouTube Video
              </span>,
            ],
            "href": "",
            "rel": "noopener noreferrer",
            "target": "_blank",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "youtube icon",
                "className": "float_left",
                "src": "",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "YouTube Video",
                "className": "youtube",
              },
              "ref": null,
              "rendered": "YouTube Video",
              "type": "span",
            },
          ],
          "type": "a",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href=""
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="wiki icon"
              className="float_left"
              src=""
            />
            <span
              className="wiki"
            >
              Wikipedia Page
            </span>
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="wiki icon"
                className="float_left"
                src=""
              />,
              <span
                className="wiki"
              >
                Wikipedia Page
              </span>,
            ],
            "href": "",
            "rel": "noopener noreferrer",
            "target": "_blank",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "wiki icon",
                "className": "float_left",
                "src": "",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Wikipedia Page",
                "className": "wiki",
              },
              "ref": null,
              "rendered": "Wikipedia Page",
              "type": "span",
            },
          ],
          "type": "a",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <a
              href=""
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="youtube icon"
                className="float_left"
                src=""
              />
              <span
                className="youtube"
              >
                YouTube Video
              </span>
            </a>
          </div>,
          <div>
            <a
              href=""
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="wiki icon"
                className="float_left"
                src=""
              />
              <span
                className="wiki"
              >
                Wikipedia Page
              </span>
            </a>
          </div>,
        ],
        "className": "links",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href=""
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="youtube icon"
                className="float_left"
                src=""
              />
              <span
                className="youtube"
              >
                YouTube Video
              </span>
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="youtube icon"
                  className="float_left"
                  src=""
                />,
                <span
                  className="youtube"
                >
                  YouTube Video
                </span>,
              ],
              "href": "",
              "rel": "noopener noreferrer",
              "target": "_blank",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "youtube icon",
                  "className": "float_left",
                  "src": "",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "YouTube Video",
                  "className": "youtube",
                },
                "ref": null,
                "rendered": "YouTube Video",
                "type": "span",
              },
            ],
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href=""
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="wiki icon"
                className="float_left"
                src=""
              />
              <span
                className="wiki"
              >
                Wikipedia Page
              </span>
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="wiki icon"
                  className="float_left"
                  src=""
                />,
                <span
                  className="wiki"
                >
                  Wikipedia Page
                </span>,
              ],
              "href": "",
              "rel": "noopener noreferrer",
              "target": "_blank",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "wiki icon",
                  "className": "float_left",
                  "src": "",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Wikipedia Page",
                  "className": "wiki",
                },
                "ref": null,
                "rendered": "Wikipedia Page",
                "type": "span",
              },
            ],
            "type": "a",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
